src/hawkbit-client.c

```diff
@@ -914,6 +914,17 @@ static gpointer download_thread(gpointer data)
         }
 
         g_mutex_lock(&active_action->mutex);
+        // skip installation if hawkBit asked us to do so
+        if (!artifact->do_install &&
+            (!artifact->do_maintenance || g_strcmp0(artifact->do_maintenance, "available") == 0)) {
+                active_action->state = ACTION_STATE_SUCCESS;
+                if (!feedback(artifact->feedback_url, active_action->id, "File checksum OK.", "success",
+                              "downloaded", &feedback_error))
+                        g_warning("%s", feedback_error->message);
```

Comment:
Wouldn't this also apply to the "normal code path" in this function? So rather replace the `if !feedback_progress(artifact->feedback_url, active_action->id, "File checksum OK.", &error)) {` below?

Reply:
Normal code path reports `proceeding` status. Or I'm missing the point here?

Reply:
Ah, I guess you're right. `success`/`downladed` only applies to the `downloadonly` case.